index.d.ts

```diff
@@ -0,0 +1,72 @@
+import * as React from 'react';
+
+
+export interface DatGuiProps {
+    data: any;                      // The data your dat.GUI controller will mutate
+    onUpdate: (data: any) => any;   // The method which will be called whenever an update is handled by the controller
+    children: React.ReactElement<any>; // The dat.GUI components that make up the controller
+    liveUpdate?: boolean;           // Determines if live updates should occur, defaults to true
+    labelWidth?: number;            // The width of the labels in pixels, defaults to 40
+    className?: string;             // The class name to set on the DatGui div
+    style?: StyleSheet;             // The style object to set on the DatGui div
+}
+
+export default class DatGui extends React.Component<DatGuiProps, any> { }
+
+export class DatUnChangableFieldProps {
+    label?: string;
+}
+
+export class DatChangableFieldProps extends DatUnChangableFieldProps {
+    path: string;
+}
+
+
+export interface DatButtonProps extends DatUnChangableFieldProps {
+    onClick: (e: React.MouseEvent) => any;
+}
+export class DatButton extends React.Component<DatButtonProps, any> { }
+
+
+export interface DatFolderProps extends DatUnChangableFieldProps {
+    title: string;
+    children: React.ReactNode[];
```

Comment:
NOTE [1]
Similar issue to 'NOTE [0]'.   Switching this to 'any' allowed me to work with the repo's provided example. https://github.com/claus/react-dat-gui/blob/master/example/src/App.js